### PR TITLE
Fetch non s3parq parquet files

### PR DESCRIPTION
**Pull request by schafrn**
What:
This PR allows S3Parq to fetch non-partitioned parquet files that weren't published with it.

Why:
This lets people have consistent handling of getting parquet files, especially since this functionality is a small-effort change to add the feature.

### CODE REVIEW

File: s3parq/fetch_parq.py [hunk 1]
```diff
@@ -171,6 +171,10 @@ def fetch(bucket: str, key: str, filters: List[type(Filter)] = {}, parallel: boo
             Determines if multiprocessing should be used, defaults to True
             Whether this is more or less efficient is dataset dependent,
                 smaller datasets run much quicker without parallel
+        accept_not_s3parq (bool, Optional):
+            Determines whether it will fetch dataframes that have not been published
+                with S3Parq and are thus lacking metadata. This does not support
+                partitioned parquet like this, only
```
Comment (RyanAdalbert) [inline s3parq/fetch_parq.py]: Did finishing this thought get forgotten?

File: s3parq/fetch_parq.py [hunk 2]
```diff
@@ -622,3 +643,12 @@ def _validate_matching_filter_data_type(part_types, filters: List[type(Filter)])
...
                     f"Comparison {f['comparison']} cannot be used on partition types of {fil_part}")
+
+
+# Custom error for catching
+class MissingS3ParqMetadata(Exception):
+    """ This class is meant to be an Exception thrown only when parquet files
+    being fetched are not S3Parq compatible and the option to fetch anyways is
+    disabled.
+    """
+    pass
```
Comment (RyanAdalbert) [inline s3parq/fetch_parq.py]: A line feed is missing at the end of this file

File: tests/test_fetch_parq.py [hunk 3]
```diff
@@ -454,3 +457,109 @@ def test_fetch_files_list_more_than_1k():
...
+        s3_client=s3_client
+    )
+
+    with pytest.raises(MissingS3ParqMetadata):
+        fetched_diff = fetch_parq.fetch(
+            bucket=input_bucket,
+            key=s3_key,
+            parallel=False,
+            accept_not_s3parq=False
+        )
```
Comment (RyanAdalbert) [inline tests/test_fetch_parq.py]: A line feed is missing at the end here

**Review by RyanAdalbert — APPROVED** [review]
Awesome Rayne, this will be a great feature to have!

File: s3parq/fetch_parq.py (hunk 1 above)
Comment (schafrn) [reply s3parq/fetch_parq.py]: Yep 😅 I fixed that sentence and added a small bit to the ReadMe, good catch!

**Pull request merged**